v3/src/components/axis/helper-models/categorical-axis-helper.ts

```diff
@@ -106,17 +109,35 @@ export class CategoricalAxisHelper extends AxisHelper {
             .attr('y2', (d, i) => isVertical
               ? fns.getDividerY(i) : (isTop ? 1 : -1) * dividerLength)
           // labels
-          update.select('.category-label')
-            .attr('transform', `${rotation}`)
-            .attr('text-anchor', textAnchor)
-            .attr('transform-origin', (d, i) => {
-              return `${fns.getLabelX(i)} ${fns.getLabelY(i)}`
-            })
-            .transition().duration(duration)
-            .attr('class', 'category-label')
-            .attr('x', (d, i) => fns.getLabelX(i))
-            .attr('y', (d, i) => fns.getLabelY(i))
-            .text((d: CatObject, i) => String(categories[i]))
+          if (isColorAxis) {
+            // Render color swatches
+            update.selectAll('text').remove()
+            update.select('.category-label')
+              .attr('class', 'category-label')
+              .attr('x', (d, i) => fns.getLabelX(i) - ((bandWidth * 2 / 3) / 2))
+              .attr('y', (d, i) => Math.max(6.5, fns.getLabelY(i) - (kDefaultColorSwatchHeight / (isVertical ? 2 : 1))))
+              .style("fill", (d, i) => categories[i])
+              .style("width", bandWidth * 2 / 3)
+              .style("height", `${kDefaultColorSwatchHeight}px`)
+              .attr('transform', `${rotation}`)
+              .attr('transform-origin', (d, i) => `${fns.getLabelX(i)} ${fns.getLabelY(i)}`)
+              .transition().duration(duration)
+              .style('opacity', 0.85)
```

Comment:
It seems like stroke and opacity would be better specified in css as they are for legend color squares